### PR TITLE
[locator] Add active layer scope to the calculator locator filter

### PR DESCRIPTION
**Pull request by nirvn**
## Description
This allows the usage of layer scope functions such as `sum("MY_FIELD")`.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [x] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [x] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by nirvn** [comment]
@nyalldawson , do we want to backport this to 3.6 and/or 3.4? I have no strong opinion.

**Review by nyalldawson — APPROVED** [review]

File: src/app/locator/qgsinbuiltlocatorfilters.cpp [hunk 1]
```diff
@@ -480,6 +480,10 @@ void QgsExpressionCalculatorLocatorFilter::fetchResults( const QString &string,
   context << QgsExpressionContextUtils::globalScope()
           << QgsExpressionContextUtils::projectScope( QgsProject::instance() );
 
+  QgsVectorLayer *layer = qobject_cast< QgsVectorLayer *>( QgisApp::instance()->activeLayer() );
```
Comment (nyalldawson) [inline src/app/locator/qgsinbuiltlocatorfilters.cpp]: No need to cast to vector here -- raster and mesh layers are ok.

Comment (nirvn) [reply src/app/locator/qgsinbuiltlocatorfilters.cpp]: I was wondering about that. OK.

**Comment by nyalldawson** [comment]
> @nyalldawson , do we want to backport this to 3.6 and/or 3.4? I have no strong opinion.

giphy-downsized-large

**Pull request merged**